cmd/util/cmd/execution-state-extract/cmd.go

```diff
@@ -119,45 +117,16 @@ func run(*cobra.Command, []string) {
 		if err != nil {
 			log.Fatal().Err(err).Msg("invalid root checkpoint")
 		}
+		defer f.Close()
 
-		const (
-			encMagicSize     = 2
-			encVersionSize   = 2
-			crcLength        = 4
-			encNodeCountSize = 8
-			encTrieCountSize = 2
-			headerSize       = encMagicSize + encVersionSize
-		)
-
-		// read checkpoint version
-		header := make([]byte, headerSize)
-		n, err := f.Read(header)
-		if err != nil || n != headerSize {
-			log.Fatal().Err(err).Msg("failed to read version from root checkpoint")
-		}
-
-		magic := binary.BigEndian.Uint16(header)
-		version := binary.BigEndian.Uint16(header[encMagicSize:])
-
-		if magic != wal.MagicBytes {
-			log.Fatal().Err(err).Msg("invalid magic bytes in root checkpoint")
-		}
-
-		if version <= 3 {
-			_, err = f.Seek(-(hash.HashLen + crcLength), 2 /* relative from end */)
-			if err != nil {
-				log.Fatal().Err(err).Msg("invalid root checkpoint")
-			}
-		} else {
-			_, err = f.Seek(-(hash.HashLen + encNodeCountSize + encTrieCountSize + crcLength), 2 /* relative from end */)
-			if err != nil {
-				log.Fatal().Err(err).Msg("invalid root checkpoint")
-			}
+		rootHash, err := wal.ReadLastTrieRootHashFromCheckpoint(f)
+		if err != nil {
+			log.Fatal().Err(err).Msgf("failed to read last root hash in root checkpoint: %s", err.Error())
 		}
 
-		n, err = f.Read(stateCommitment[:])
-		if err != nil || n != hash.HashLen {
-			log.Fatal().Err(err).Msg("failed to read state commitment from root checkpoint")
+		stateCommitment, err = flow.ToStateCommitment(rootHash[:])
```

Comment:
may I request a brief update / extension of `ToStateCommitment`'s goDoc: 
https://github.com/onflow/flow-go/blob/8a2063475ac150fecd883950dbda81153357a7ff/model/flow/ledger.go#L111-L113
* Please correct the typo `StateComitment` -> `StateCommitment` 
* Please include a brief description of the expected error returns. Something like the following would already suffice:
    ```golang
    // All returned errors indicate that the given byte slice is not a valid root hash of an execution state.
    // As the function is side-effect free, all failures are simply a no-op.
    ```

Reply:
Thanks! 👍   Done in commit 7d8cba1.